model-sync-gradle-test/src/test/kotlin/org/modelix/model/sync/gradle/test/PullTest.kt

```diff
@@ -0,0 +1,76 @@
+                    <child id="466301921131663001" name="nodes" index="1DmyQT" />
+                    <child id="466301921131630381" name="edges" index="1DmEKd" />
+                  </concept>
+                  <concept id="466301921131629396" name="GraphLang.structure.Node" flags="ng" index="1DmExO" />
+                  <concept id="466301921131629399" name="GraphLang.structure.Edge" flags="ng" index="1DmExR">
+                    <reference id="466301921131663029" name="source" index="1DmyQl" />
+                    <reference id="466301921131663030" name="target" index="1DmyQm" />
+                  </concept>
+                </language>
+                <language id="ceab5195-25ea-4f22-9b92-103b95ca8c0c" name="jetbrains.mps.lang.core">
+                  <concept id="1169194658468" name="jetbrains.mps.lang.core.structure.INamedConcept" flags="ng" index="TrEIO">
+                    <property id="1169194664001" name="name" index="TrG5h" />
+                  </concept>
+                </language>
+              </registry>
+              <node concept="1DmExM" id="pSCM1J8y9y">
+                <node concept="1DmExR" id="pSCM1J8Fg1" role="1DmEKd">
+                  <ref role="1DmyQl" node="pSCM1J8FfW" resolve="X" />
+                  <ref role="1DmyQm" node="pSCM1J8FfX" resolve="Y" />
+                </node>
+                <node concept="1DmExR" id="pSCM1J8Fg2" role="1DmEKd">
+                  <ref role="1DmyQl" node="pSCM1J8FfX" resolve="Y" />
+                  <ref role="1DmyQm" node="pSCM1J8FfY" resolve="Z" />
+                </node>
+                <node concept="1DmExR" id="pSCM1J8Fg3" role="1DmEKd">
+                  <ref role="1DmyQl" node="pSCM1J8FfZ" resolve="D" />
+                  <ref role="1DmyQm" node="pSCM1J8FfY" resolve="Z" />
+                </node>
+                <node concept="1DmExR" id="pSCM1J8Fg4" role="1DmEKd">
+                  <ref role="1DmyQl" node="pSCM1J8Fg0" resolve="E" />
+                  <ref role="1DmyQm" node="pSCM1J8FfZ" resolve="D" />
+                </node>
+                <node concept="1DmExO" id="pSCM1J8FfW" role="1DmyQT">
+                  <property role="TrG5h" value="X" />
+                </node>
+                <node concept="1DmExO" id="pSCM1J8FfX" role="1DmyQT">
+                  <property role="TrG5h" value="Y" />
+                </node>
+                <node concept="1DmExO" id="pSCM1J8FfY" role="1DmyQT">
+                  <property role="TrG5h" value="Z" />
+                </node>
+                <node concept="1DmExO" id="pSCM1J8FfZ" role="1DmyQT">
+                  <property role="TrG5h" value="D" />
+                </node>
+                <node concept="1DmExO" id="pSCM1J8Fg0" role="1DmyQT">
+                  <property role="TrG5h" value="E" />
+                </node>
+              </node>
+            </model>
+        """.trimIndent()
+
+        assertEquals(expected, localModelFile.readText().trim())
```

Comment:
Is it valid to assume that the exact XML format is always created, without spacing changes etc.? Otherwise I'd propose to use something like [xmlunit](https://www.xmlunit.org/) to be safe from such issues.

Reply:
And we are sure that the formatting doesn't change on the next MPS bump?